### ISSUE TITLE
Comp Processor Processing Unchanged Data between Edits?

### ISSUE THREAD

**Issue by msweier** [opendcs/opendcs#743]
**Describe the bug**
It appears that CP is recomputing unchanged data in between two data points that have been edited. I thought this was a CWMS-VUE behavior, but using cwms-python/CDA, I'm thinking it's an OpenDCS behavior. 

**To Reproduce**
I wrote null values to two data spikes about 1 month apart. 
![image](https://github.com/user-attachments/assets/5b41709f-e3be-45b7-8579-ee49c255f8d4)
And it triggered 9,100 comps. 
![image](https://github.com/user-attachments/assets/25a12531-74aa-4ebc-88c1-330290fba6d7)

**Expected behavior**
I would expect maybe 50 computations at most for two data points considering daily average etc.... I would expect it would only queue the data that was corrected....not the data in between too. 

**Desktop/ServerSide (please complete the following information):**
OpenDCS 7.0.12

**Additional context**

**Comment by MikeNeilson** [comment]
That's odd. A gap of more than 7 days should get pretty well short cut. I'll have to check the logic in that retrieval.

**Comment by msweier** [comment]
I tried adding `cp_cwmstsdb_getNewData_max_timegap_days=7` to the user properties file and restarted as suggested, but I'm still getting the same behavior. Some have suggested that the gap between start and end points are getting filled in by the API CDA uses.

**Comment by MikeNeilson** [comment]
Hmm. if you don't mind doing a bit of extra work you can try putting the data in the format of the linked file and calling `importts <filename>`

https://github.com/opendcs/opendcs/blob/main/integrationtesting/opendcs-tests/src/test/resources/data/timeseries/OpenDCS-Postgres/regular_ts.tsimport

This would bypass CDA. I don't CDA is the issue here though, it *might* be the database "filling" in the gap when things get added to the task list.


That's actually another thing to check.
1. Turn off the comp
2. Save the data
3. Wait a moment
4. Lookup that ts_code in the tasklist.

If the task list has more than the two values, it could still be OpenDCS, but it would be an in database method.

**Comment by msweier** [comment]
Ok importing using impportts two values about a month and a half apart for a site yields 3875 comps. Seems like maybe it's something on the DB end then.

**Comment by MikeNeilson** [comment]
Yeah, could be compproc behavior, could be the tasklist. Will have to experiment more.

**Comment by msweier** [comment]
Ok let me know if there are any more tests I can do to help. This issue is a sticking point in our district w/ our setup. Thanks!

**Comment by MikeNeilson** [comment]
If you have the time to turn off the compproc instance, save the data, and check the tasklist that would be useful. Isolates to one of the places to look at behavior.

**Comment by msweier** [comment]
In my test I did turn off the process that was running that comp when I saved the data and got the same behavoir. I assume that's the same as the compproc instance?

**Comment by MikeNeilson** [comment]
... wait, did you click save in CompEdit? If the compproc instance was off, no data should've processed.

Or do you mean there *was* a record in `select * from cp_comp_tasklist` for all the values between the two times?

**Comment by msweier** [comment]
Here's what I did:

1. Ran the SQL query to make sure the tasklist was near zero.
2. Turned off the process that was running the comp I knew would be triggered by data that was edited with the `stopcomp` in command line. 
3. Pushed two missing values about 40 days apart into the database with the `importts fileName` command.
4. Ran the SQL query which showed 3875 comps associated with the timeseries I just saved.
5. Sighed and turned the process back on.

**Comment by MikeNeilson** [comment]
Ah. That definitely narrows it down to either the message that data was saved, or the way it was processed.

**Comment by MikeNeilson** [comment]
Yep, that'll do it:

```
insert into cp_comp_tasklist
      (  record_num,
         loading_application_id,
         site_datatype_id,
         date_time_loaded,
         start_date_time,
         value,
         unit_id,
         delete_flag,
         quality_code,
         flags)
      select
         cp_comp_tasklistidseq.nextval,
         r2.loading_application_id,
         p_ts_code,
         sysdate,
         r1.date_time,
         nvl(r1.value,0),
         l_unit_id,
         decode (r1.quality_code,5,'Y','N') delete_flag,
         r1.quality_code,
         r1.quality_code
      from cwms_v_tsv r1
      where r1.ts_code = p_ts_code 
        and r1.date_time >= l_start_time and r1.date_time <= l_end_time -- <----- **** THIS BIT RIGHT HERE ***, says get me all the things.
        and r1.data_entry_date <= p_enqueue_time;
    end loop;   /* end of for r2 loop */
```

**Comment by MikeNeilson** [comment]
Now I think the fix to that is actually really simple, you know, just only get the values presented, and I can hot patch that so we can find out, but let's do that on Monday. not minutes before I leave for the week.

**Comment by MikeNeilson** [comment]
No, it would be an easy fix if the TSDataStored message wasn't a range.

That said, I think it's still relatively easy. Given the data in the middle won't have the same data entry date I think a simple conditional on that should do the trick.

@perrymanmd seem reasonable to you? Changing the message itself seems a bit... much. And it looks like the "store_time" value is the same as data_entry_date just with a different name. If so makes the query even easier.

**Comment by perrymanmd** [comment]
Yes the message store_date is the same as the data_entry_date in AT_TSV_XXXX

**Comment by MikeNeilson** [comment]
Awesome. Thanks!

**Comment by MikeNeilson** [comment]
Yeah, I believe current version of CWMSVue still forcibly overwrite everything, thus updating the "data_entry_date", this should be changed in the next release of that software though - I think.
